internal/parameterizer/parameterizer_test.go

```diff
@@ -0,0 +1,247 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parameterize_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/google/go-cmp/cmp"
+
+	common "github.com/konveyor/move2kube/internal/common"
+	parameterize "github.com/konveyor/move2kube/internal/parameterizer"
+	"github.com/konveyor/move2kube/internal/types"
+	irtypes "github.com/konveyor/move2kube/internal/types"
+	plantypes "github.com/konveyor/move2kube/types/plan"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestParameterizer(t *testing.T) {
+	log.SetLevel(log.DebugLevel)
+
+	t.Run("1.IR with no services, no storage", func(t *testing.T) {
+
+		ir := getIRWithoutServices()
+		irAfter, err := parameterize.Parameterize(ir)
+
+		if err != nil {
+			t.Fatal("Failed to parameterize. Error:", err)
+		}
+
+		irAfter.TargetClusterSpec.Host = ""
+
+		if !cmp.Equal(ir, irAfter, cmpopts.EquateEmpty()) {
+			t.Fatalf("TODO explanation. Difference:\n%s", cmp.Diff(ir, irAfter))
+		}
+
+	})
+
+	/*
+		t.Run("2.IR containing services that have no containers", func(t *testing.T) {
+
+			ir := getIRWithServicesAndWithoutContainers()
+			irAfter, err := parameterize.Parameterize(ir)
+			if err != nil {
+				t.Fatal("Failed to parameterize. Error:", err)
+			}
+
+			// oracle
+			// if the service has no containers, then no changes will be done
+			// ir = irAfter
+			if !cmp.Equal(ir, irAfter, cmpopts.EquateEmpty()) {
+				t.Fatalf("TODO explanation. Difference:\n%s", cmp.Diff(ir, irAfter))
+			}
+
+		})
+	*/
+
+	t.Run("3.IR containing services with containers", func(t *testing.T) {
+
+		ir := getIRWithServicesAndContainers()
+		irAfter, err := parameterize.Parameterize(ir)
+		if err != nil {
+			t.Fatal("Failed to parameterize. Error:", err)
+		}
+		fmt.Printf("%T", irAfter)
+		// oracle
+
+	})
+
+	t.Run("4.IR with no services , but storage, storage type is not PVCKind ", func(t *testing.T) {
+
+		// as the type is not PVCKind, the scMap will be empty
+		// -> no parameterizations -> before = after
+		ir := getIRWithStorageNotPVCKind()
+		irAfter, err := parameterize.Parameterize(ir)
+		if err != nil {
+			t.Fatal("Failed to parameterize. Error:", err)
+		}
+
+		// oracle
+
+		irAfter.TargetClusterSpec.Host = ""
+		if !cmp.Equal(ir, irAfter, cmpopts.EquateEmpty()) {
+			t.Fatalf("TODO explanation. Difference:\n%s", cmp.Diff(ir, irAfter))
+		}
+
+	})
+
+	t.Run("5.IR with no services , but storage, storage type is PVCKind ", func(t *testing.T) {
+
+		// scMap should be 1, we can parameterize
+		ir := getIRWithStoragePVCKind()
+		irAfter, err := parameterize.Parameterize(ir)
+		if err != nil {
+			t.Fatal("Failed to parameterize. Error:", err)
+		}
+
+		// oracle:
+		paramSC := "{{ .Values.storageclass }}"
+		for _, j := range irAfter.Storages {
+			x := *j.PersistentVolumeClaimSpec.StorageClassName
+			if x != paramSC {
```

Comment:
Since the function takes the entire `ir` struct as input and produces a new `ir` struct, we should be doing full comparison with `cmp.Equal(actual, want)`. We need to make sure all the fields are as expected.